src/components/select/select-list/select-list.component.js

```diff
@@ -0,0 +1,170 @@
+import React, {
+  useEffect,
+  useState,
+  useCallback
+} from 'react';
+import PropTypes from 'prop-types';
+import StyledSelectList from './select-list.style';
+import Events from '../../../utils/helpers/events/events';
+import updateListScrollTop from './update-list-scroll';
+import getNextChildByText from '../utils/get-next-child-by-text';
+import Portal from '../../portal/portal';
+import { getNextIndexByKey } from '../utils/get-next-child-by-key-code';
+
+const SelectList = React.forwardRef(({
+  id,
+  children,
+  onSelect,
+  onSelectListClose,
+  filterText,
+  anchorElement,
+  highlightedValue
+}, listRef) => {
+  const [currentOptionsListIndex, setCurrentOptionsListIndex] = useState(-1);
+
+  const highlightItem = useCallback((newIndex, optionList) => {
+    const { text, value } = optionList[newIndex].props;
+
+    setCurrentOptionsListIndex(newIndex);
+    updateListScrollTop(newIndex, listRef.current);
+    onSelect({ text, value, selectionType: 'navigationKey' });
+  }, [listRef, onSelect]);
+
+  const handleGlobalKeydown = useCallback((event) => {
+    const optionList = React.Children.toArray(children);
+    const lastIndex = optionList.length - 1;
+
+    if (Events.isTabKey(event) || Events.isEscKey(event)) {
+      onSelectListClose();
+    } else if (Events.isEnterKey(event)) {
+      const currentOption = optionList[currentOptionsListIndex];
+
+      if (!currentOption) {
+        onSelectListClose();
+
+        return;
+      }
+
+      const { text, value } = currentOption.props;
+
+      onSelect({ text, value, selectionType: 'enterKey' });
+    } else if (isNavigationKey(event)) {
+      const nextIndex = getNextIndexByKey(event.which, currentOptionsListIndex, lastIndex);
```

Comment:
`event.which` is deprecated https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/which

You probably want `event.key` instead.